test/e2e_node/swap_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	nodev1 "k8s.io/api/node/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubefeatures "k8s.io/kubernetes/pkg/features"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	reservedSwapSize      = "256Mi"
+	reversedSwapSizeBytes = 2 << 27
+)
+
+// Serial because the test updates kubelet configuration.
+var _ = SIGDescribe("System reserved swap [Serial]", func() {
+	f := framework.NewDefaultFramework("system-reserved-swap")
+	ginkgo.Context("With config updated with swap reserved", func() {
+		tempSetCurrentKubeletConfig(f, func(initialConfig *kubeletconfig.KubeletConfiguration) {
+			initialConfig.FailSwapOn = false
+			initialConfig.FeatureGates[string(kubefeatures.NodeSwap)] = true
```

Comment:
We configure Swap in the test environment - given that it also requires special OS config, I'm not sure we should try enabling it in the test

Reply:
If swap is not enabled on the OS, we should skip the test case.

https://github.com/kubernetes/kubernetes/pull/105271/files#diff-40c4fc8e2327888a122bfcb2f8e7232e0164639a3f2da621dd4d9a70b1eb1ee4R67-R69

Line 67-69 will check that and skip if needed.